packages/snaps-controllers/src/snaps/endowments/name-lookup.ts

```diff
@@ -128,20 +156,53 @@ export function getChainIdsCaveat(
     return null;
   }
 
-  assert(permission.caveats.length === 1);
-  assert(permission.caveats[0].type === SnapCaveatType.ChainIds);
+  const caveat = permission.caveats.find(
+    (permCaveat) => permCaveat.type === SnapCaveatType.ChainIds,
+  );
+  assert(permission.caveats.length === 1 || permission.caveats.length === 2);
+  assert(caveat);
+  assert(caveat.type === SnapCaveatType.ChainIds);
+
+  return (caveat as Caveat<string, string[]>).value ?? null;
+}
 
-  const caveat = permission.caveats[0] as Caveat<string, string[]>;
+/**
+ * Getter function to get the matchers caveat from a permission.
+ *
+ * This does basic validation of the caveat, but does not validate the type or
+ * value of the namespaces object itself, as this is handled by the
+ * `PermissionsController` when the permission is requested.
+ *
+ * @param permission - The permission to get the `matchers` caveat from.
+ * @returns A `matchers` object that defines the input that the snap supports.
+ */
+export function getLookupMatchersCaveat(
+  permission?: PermissionConstraint,
+): Record<string, string[]> | null {
+  if (!permission?.caveats) {
+    return null;
+  }
 
-  return caveat.value ?? null;
+  const caveat = permission.caveats.find(
+    (permCaveat) => permCaveat.type === SnapCaveatType.LookupMatchers,
+  );
+  assert(permission.caveats.length === 1 || permission.caveats.length === 2);
```

Comment:
Is this assertion still needed?

Reply:
I suppose not.

Reply:
This has not been resolved for `getLookupMatchersCaveat`.